### PR TITLE
Fix Windows ARM64 Detection

### PR DESCRIPTION
**Pull request by hmelder**
CMAKE_SYSTEM_PROCESSOR returns ARM64 and not aarch64 (cmake version 3.27.9).

systeminfo:
```powershell
OS Name:                   Microsoft Windows 11 Pro
OS Version:                10.0.22598 N/A Build 22598
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
System Type:               ARM64-based PC
```

### CODE REVIEW

File: cmake/modules/DispatchWindowsSupport.cmake [hunk 1]
```diff
@@ -6,7 +6,7 @@ function(dispatch_windows_arch_spelling arch var)
     set(${var} x64 PARENT_SCOPE)
   elseif(${arch} STREQUAL armv7)
     set(${var} arm PARENT_SCOPE)
-  elseif(${arch} STREQUAL aarch64)
+  elseif(${arch} STREQUAL aarch64 OR ${arch} STREQUAL ARM64)
```
[the remark this answers is not in the document]
Comment (hmelder) [reply cmake/modules/DispatchWindowsSupport.cmake:9]: I am running on a prerelease version from late 2022, maybe they changed the architecture name later on.

> Why not use MATCHES?

I was just adapting the method used for x64 and AMD64 detection. Should I change it?

**Comment by triplef** [comment]
Would there be a simple way to also support cross-compiling on Windows using the respective VS environments e.g. by looking at the `VSCMD_ARG_TGT_ARCH` env var instead of `CMAKE_SYSTEM_PROCESSOR`? Not sure what’s the common CMake thing to do on Windows, but personally I would expect it to build for whatever VS environment I am using.

**Comment by hmelder** [comment]
@compnerd Given the previous use of `OR` instead of `MATCHES` how should I move on with this?

**Comment by compnerd** [comment]
Eh, I suppose that it is fine as is.

**Comment by compnerd** [comment]
@swift-ci please test

**Review by compnerd — APPROVED** [review]

**Comment by compnerd** [comment]
Hmm, actually, is this still needed after #812?

**Comment by hmelder** [comment]
> Hmm, actually, is this still needed after https://github.com/apple/swift-corelibs-libdispatch/pull/812?

Nope

**Pull request closed without merging**